flake-module.nix

```diff
@@ -0,0 +1,26 @@
+{ self, lib, flake-parts-lib, ... }:
+let
+  inherit (flake-parts-lib)
+    mkPerSystemOption;
+  inherit (lib)
+    mkOption
+    types;
+in
+{
+  options = {
+    perSystem = mkPerSystemOption
+      ({ config, self', inputs', pkgs, system, ... }: {
+        options.treefmt = mkOption {
+          description = ''
+            Project-level treefmt configuration
+          '';
+          type = types.submoduleWith {
+            modules = (import ./.).all-modules pkgs;
+          };
+        };
+        config = {
+          checks.treefmt = config.treefmt.build.check self;
+        };
+      });
+  };
+}
```

Comment:
I like how smaller this module has become. Nice :)

Maybe a last question to @roberth mostly; given that flake-parts has a global namespace, do you think we should prefix treefmt into something like `programs.treefmt` for example?

Reply:
I don't mind a flat namespace at all. And maybe treefmt is more than a program?
In Nixpkgs we're actually trying to get rid of the categorization hierarchy for various reasons. RFC coming soon, but preparations are happening here https://github.com/nixpkgs-architecture/simple-package-paths